contrib/lkt/language/parser.py

```diff
@@ -1843,7 +1843,10 @@ def is_authorized_decl(decl=T.Decl.entity):
         pass
 
     @langkit_property()
-    def check_correctness_pre():
+    def check_legality():
+        # TODO: This implementation implies that we'll run full checking for
+        # all decls, even if they're in a context where we don't want to have
+        # them. Is that a good idea?
```

Comment:
I don’t understand this: if I understand correctly the new organization for checks, `check_correctness_pre` is the called for all nodes while `check_legality` is called for nodes in a decl only when only when name resolution/typing validated that decl. What context are you referring to? Also can you expand the commit message to explain what’s the purpose of moving the checks?

Reply:
This comment is in the implementation of whether a decl is authorized in a certain context or not. Since we do that as a semantic check, and do not exempt further resolution, I added this comment because I think it might happen that having a declaration in a context where it is not authorized might generate further message, or in other words, we might want to exempt the whole decl from further messages.

I moved it because so far it has no functional reason to be in `check_correctness_pre`, and the whole point of this refactoring was to have nothing in there that you can avoid having there. I think this is plenty explained in comments in this PR.